### PR TITLE
Autocomplete feature doc

### PR DESCRIPTION
**Pull request by Arsnael**
Doc:
 - [x] Feature description
 - [x] JMAP endpoint
 - [x] Webadmin additions
 - [x] How to run LSC scripts to sync contacts from a LDAP
 - [x] AMQP endpoint to sync a contact application

### CODE REVIEW

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/contactAutocomplete.adoc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+			  }, {
+                  "id": "4",
+                  "firstname": "Marie",
+                  "surname": "Dupond",
+                  "emailAddress": "mdupond@linagora.com"
+              }]
+		  },
+		  "c1"
+    ]
+  ]
```
Comment (chibenwa) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/contactAutocomplete.adoc:112]: Ident looks weird.

**Review by chibenwa — APPROVED** [review]
Good so far ;-)

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/contactAutocomplete.adoc (hunk 1 above)
Comment (Arsnael) [reply docs/modules/ROOT/pages/tmail-backend/jmap-extensions/contactAutocomplete.adoc:112]: I keep having weird stuff on Intellij with indents recently... will double check

File: docs/modules/ROOT/pages/tmail-backend/webadmin.adoc [hunk 2]
```diff
@@ -406,4 +406,125 @@ Return codes:
...
+
+Return codes:
+
+- `200` Retrieve the list of contacts successfully
+- `400` Invalid domain
+
+=== List all contacts from all domains
+
+....
+curl -XGET http://ip:port/domains/contacts
```
Comment (quantranhong1999) [inline docs/modules/ROOT/pages/tmail-backend/webadmin.adoc:519]:
```suggestion
curl -XGET http://ip:port/domains/contacts/all
```

**Review by quantranhong1999 — APPROVED** [review]

File: docs/modules/ROOT/pages/tmail-backend/webadmin.adoc (hunk 2 above)
Comment (Arsnael) [reply docs/modules/ROOT/pages/tmail-backend/webadmin.adoc:519]: Good catch :D

**Review by quantranhong1999 — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by vttranlina — APPROVED** [review]

File: docs/modules/ROOT/pages/tmail-backend/features/contactAutocomplete.adoc [hunk 3]
```diff
@@ -48,7 +48,89 @@ link:https://github.com/linagora/tmail-lsc#domain-contact-synchronization[domain
...
+
+You will need to configure TMail to connect to a RabbitMQ instance that will consume messages sent by the third-party app
+on a dedicated queue.
+
+In your `rabbitmq.properties` file in TMail, you need to add those extra fields:
+
+- `address.contact.uri`: the URI of RabbitMQ having messages related to contacts from the third-party app.
+- `address.contact.user`: the username to access the RabbitMQ instance.
+- `address.contact.password`: the password to access the RabbitMQ instance.
+- `address.contact.exchange`: the exchange name for contacts on RabbitMQ.
```
Comment (vttranlina) [inline docs/modules/ROOT/pages/tmail-backend/features/contactAutocomplete.adoc:61]: As the last commit https://github.com/linagora/tmail-backend/pull/405/commits/504768a7c076cdd185a4c475bb5bdcd312ba2985, 
We don't need `address.contact.exchange`

Comment (quantranhong1999) [inline docs/modules/ROOT/pages/tmail-backend/features/contactAutocomplete.adoc:61]: A reminder to sync with Tung work a bit here.
https://github.com/linagora/tmail-backend/pull/405/commits/504768a7c076cdd185a4c475bb5bdcd312ba2985

Comment (quantranhong1999) [reply docs/modules/ROOT/pages/tmail-backend/features/contactAutocomplete.adoc:61]: duplicated

Comment (Arsnael) [reply docs/modules/ROOT/pages/tmail-backend/features/contactAutocomplete.adoc:61]: Yeah I noticed... I'm just waiting to see your work being merged now than keeping modifying it with each commit :D

**Review by chibenwa — APPROVED** [review]

**Pull request merged**